python/lsst/ts/Dome/dome_csc.py

```diff
@@ -0,0 +1,142 @@
+# import asyncio
+import pathlib
+from lsst.ts import salobj
+
+__all__ = ['DomeCsc']
+
+
+class DomeCsc(salobj.ConfigurableCsc):
+    """
+    Upper level Commandable SAL Component to interface with the LSST Dome lower level components.
+    """
+
+    def __init__(
+            self,
+            config_dir=None,
+            initial_state=salobj.State.STANDBY,
+            simulation_mode=0,
+    ):
+        schema_path = (
+            pathlib.Path(__file__)
+            .resolve()
+            .parents[4]
+            .joinpath("schema", "Dome.yaml")
+        )
+
+        self.reader = None
+        self.writer = None
+        self.config = None
+
+        self.mock_ctrl = None  # mock controller, or None if not constructed
+        super().__init__(
+            name="DCS",
+            index=0,
+            schema_path=schema_path,
+            config_dir=config_dir,
+            initial_state=initial_state,
+            simulation_mode=simulation_mode,
+        )
+
+    async def do_Dome_command_crawlAz(self, id_data):
```

Comment:
You should only keep the last part of the command name. 

`Dome_command_crawlAz` -> `do_crawlAz`

Same for the other commands too.

Reply:
You may also rename the argument `id_data` to `data`, e.g.;

https://github.com/lsst-ts/ts_salobj/blob/f242ce43053635056362f034c8afd544b87c1119/python/lsst/ts/salobj/base_csc.py#L286

The `id_data` was the name on an older version of salobj.

Reply:
Thanks, both done.

Reply:
To follow up on this a bit: salobj CSCs have the following attributes:
* `cmd_<x>`: a `salobj.topics.ControllerCommand` for command <x>, such as `cmd_standby`. You will rarely need to do anything with these.
* `do_<x>` to implement command <x>. salobj automatically assigns these methods as callback functions to the commands `cmd_<x>`.
* evt_<x>: a salobj.topics.ControllerEvent for event <x>, e.g. evt_summaryState.
* tel_<x>: a salobj.topics.ControllerTelemetry for telemetry topic <x>.

The CSC name and SAL terms "command" and "logevent" are not part of these names. You can consider "evt" to be shorthand for "logevent" and "cmd" to be shorthand for "command", but telemetry topics don't have anything like that (which is one reason I departed from SAL on this. I also wanted to make the prefixes the same length and short enough to be unobtrusive).